client/lib/products-values/index.js

```diff
@@ -166,13 +168,27 @@ export function isJetpackMonthlyPlan( product ) {
 	return isMonthly( product ) && isJetpackPlan( product );
 }
 
-export function isMonthly( product ) {
-	product = formatProduct( product );
+export function isMonthly( rawProduct ) {
+	const product = formatProduct( rawProduct );
 	assertValidProduct( product );
 
 	return parseInt( product.bill_period, 10 ) === PLAN_MONTHLY_PERIOD;
 }
 
+export function isYearly( rawProduct ) {
+	const product = formatProduct( rawProduct );
+	assertValidProduct( product );
```

Comment:
unrelated to this PR, but I think @gwwar may have brought it up, but we may want to earmark `assertValidProduct` for iteration later so that we run it in dev-only environments. throwing an error may have some use because we can see in our error-reporter if it's happening in production, but on the other hand it's running and shipping code we might not want to and crashing the execution while customers are using the app

fun fact! if we surround anything in this test it will be entirely stripped from the code in production

```js
if ( 'production' !== process.env.NODE_ENV ) {
	console.log( 'this only exists in non-prod environments' );
}
```